src/pallet/stake-exchange/processors/offer-created.ts

```diff
@@ -39,8 +45,58 @@ function getFilterFromType(tokenFilter: TokenFilter | undefined) {
     return entity
 }
 
+async function savePoolsOffersForStakeExchange(
+    event: OfferCreated,
+    ctx: CommonContext,
+    offer: StakeExchangeOffer
+): Promise<void> {
+    if (event.offer.tokenFilter?.__kind === StakeExchangeTokenFilterType.Whitelist) {
+        for (const value of event.offer.tokenFilter.value) {
+            const pool: NominationPool | undefined = await ctx.store.findOne(NominationPool, {
+                where: {
+                    id: value.toString(),
+                    state: PoolState.Open,
+                },
+            })
+
+            if (pool) {
+                const poolsOffers = new PoolsOffers({
+                    offer,
+                    pool,
+                })
+                await ctx.store.save(poolsOffers)
+            }
+        }
```

Comment:
you can find pools by doing where: { id: In(event.offer.tokenFilter.value) }
no need to query by each pool